async_test.go

```diff
@@ -170,3 +171,33 @@ func TestAddMongoDBFilesToCacheWithBadPath(t *testing.T) {
 	require.NoError(t, err)
 	assert.Error(t, addMongoDBFilesToCache(lru.NewCache(), absPath)("foo.txt"))
 }
+
+func TestDoDownloadWithValidInfo(t *testing.T) {
+	file, err := ioutil.TempFile("build", "out.txt")
+	require.NoError(t, err)
+	defer os.Remove(file.Name())
+
+	info := DownloadInfo{
+		URL:  "https://example.com",
+		Path: file.Name(),
+	}
+	req, err := http.NewRequest(http.MethodGet, info.URL, nil)
+	require.NoError(t, err)
+
+	assert.NoError(t, DoDownload(req, info, http.Client{}))
```

Comment:
If this should write a thing to the filesystem, should you check that it really did?

Reply:
Added a check for non-empty file.